community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -289,6 +289,12 @@ default void recoveryCompleted( long numberOfPagesVisited, long numberOfCleanedC
     private final Monitor monitor;
 
     /**
+     * Whether or not this tree has been closed. Accessed and changed solely in
+     * {@link #close()} and {@link #close(Consumer)} to be able to close tree multiple times gracefully.
+     */
+    private volatile boolean closed;
```

Comment:
Best-case, the lock is an `unsafe.compareAndSwapInt` which has a javadoc mentioning that it has memory semantics of a volatile read and write. So I'd say it's not necessary to have it volatile. It's a minor thing and only checked in close, but can do this change anyway.